src/pages/questions/useQuestionBuffer.ts

```diff
@@ -87,21 +143,58 @@ function reducer(state, action) {
         answers: [
           ...state.answers,
           {
-            insight_id: answeredQuestion?.insight_id,
-            barcode: answeredQuestion?.barcode,
-            insight_type: answeredQuestion?.insight_type,
-            value: answeredQuestion?.value,
+            ...answeredQuestion,
             validationValue: action.payload.value,
+            isPending: true,
+            sendingTime: new Date().getTime() + action.payload.pendingDelay,
           },
         ],
         remainingQuestionNb: state.remainingQuestionNb - 1,
         // skipped ids is used to correctly compute remainingQuestionNb after each data fetching
         skippedIds: [
           ...state.skippedIds,
-          ...(action.payload.value === -1 ? [action.payload.insight_id] : []),
+          ...(action.payload.value === -1 ? [action.payload.insightId] : []),
         ],
       };
 
+    case "sendAnswers":
+      const minDate = new Date().getTime();
+
+      const newAnswers = state.answers.map((answer) => {
```

Comment:
what about filtering the ones that need to be annotated, and then doing a `Promise.all( filtered( item => robotoff.annotate())).then(res => return { ... }).catch(()=>{})` that way all the annotations are grouped,  I feel that doing the `robotoff.annotate` inside the map is doing two different things at the same time

Reply:
I prefer doing it in the map, because like that I send the request and set `isPending` to `false` at the same time.

For now, their is no verification if the request returned an error or not. So I have no await in this code